compiler/rustc_ast/src/token.rs

```diff
@@ -314,15 +314,24 @@ pub enum TokenKind {
     Literal(Lit),
 
     /// Identifier token.
-    /// Do not forget about `NtIdent` when you want to match on identifiers.
+    /// Do not forget about `InterpolatedIdent` when you want to match on identifiers.
     /// It's recommended to use `Token::(ident,uninterpolate,uninterpolated_span)` to
     /// treat regular and interpolated identifiers in the same way.
     Ident(Symbol, IdentIsRaw),
+    /// This identifier (and its span) is the identifier passed to the
+    /// declarative macro. The span in the surrounding `Token` is the span of
+    /// the `ident` metavariable in the macro's RHS.
+    InterpolatedIdent(Ident, IdentIsRaw),
```

Comment:
```suggestion
    NtIdent(Ident, IdentIsRaw),
```
Maybe keep the `NtIdent`/`NtLifetime` naming for brevity?

(Semantically, both "interpolated" and "nonterminal" names don't make much sense to me in the current setup, they are just used as an ancient tradition.)

Reply:
I agree that "interpolated" and "nonterminal" are not very meaningful.

The reason I chose `Interpolated{Ident,Lifetime}` is because it pairs well with the `Token::uninterpolate` function, which converts them to `{Ident,Lifetime}`. When I see something like `match token.uninterpolate().kind`, it helps me remember exactly what is happening.

But I can change to `NtIdent/NtLifetime` if you feel strongly about it.

Reply:
FWIW, I planned to remove all the "uninterpolate" stuff as well after landing https://github.com/rust-lang/rust/pull/119412.

Reply:
Ok, I'll switch to `NtIdent`/`NtLifetime`.